src/target/riscv/riscv-013.c

```diff
@@ -547,7 +563,19 @@ static int dmi_op_timeout(struct target *target, uint32_t *data_in, int dmi_op,
 	while (1) {
 		status = dmi_scan(target, &address_in, data_in, DMI_OP_NOP, address, 0,
 				false);
-		if (status == DMI_STATUS_BUSY) {
+		if (status == 3 && address_in == (1U << info->abits) - 1 &&
+				*data_in == 0xffffffff) {
+			LOG_ERROR("TDO seems to be stuck high; target might be held in "
+					"reset, or there might be a connection failure");
+			return ERROR_FAIL;
+		} else if (dmi_op == DMI_OP_READ && address != 0 && status == 0 &&
+				address_in == 0 && *data_in == 0) {
+			/* On read ops we should at least get back the address as non-zero.
```

Comment:
Is that actually required by the spec? Or just happens to be the way the implementations we have do it (there is a proposed PR in rocket-chip which would actually break this assumption for writes, which is why I ask).

Reply:
Good point. It's not required by the spec.